### PR TITLE
Add field to /api/nodes/wg-net/<network id>/creategateway for enabling/disabling NAT

### PR DESCRIPTION
**Pull request by Exchizz**
Hi,

This PR adds a field to `/api/nodes/wg-net/<network id>/creategateway` for enabling/disabling NAT for egress gateways.

Example body of a POST request:
```
{
  "ranges": [
    "1.2.3.0/24"
  ],
  "interface": "eth0",
  "natEnabled": "yes"
}
```

If "NatEnabled" is left out, it will default to "yes" hence work as before this PR.
If set to "no", it will not create the iptable rules for adding/deleting the iptables MASQUERADE rule.

I have not added any tests, let me know if you want me to.

EDIT: Tests added

\Mathias

### CODE REVIEW

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Comment by 0xdcarns** [comment]
A test to ensure the code doesn't affect previous behavior would be nice to have. 

i.e. ensure by default, it is enabled

File: models/structs.go [hunk 1]
```diff
@@ -149,6 +149,7 @@ type EgressGatewayRequest struct {
 	NodeID      string   `json:"nodeid" bson:"nodeid"`
 	NetID       string   `json:"netid" bson:"netid"`
 	RangeString string   `json:"rangestring" bson:"rangestring"`
+	NatEnabled  bool     `josn:"natEnabled" bson:"natEnabled"`
```
[the remark this answers is not in the document]
Comment (Exchizz) [reply models/structs.go]: Fixed

**Comment by Exchizz** [comment]
> A test to ensure the code doesn't affect previous behavior would be nice to have.
> 
> i.e. ensure by default, it is enabled

I've added three tests.
1. tests if "-J MASQUERADE" is present in postUp/postDown if NatEnabled is "yes"
2. tests if "-J MASQUERADE" is not present in postUp/PostDown if NatEnabled is "no"
3. tests if "-J MASQUERADE" is present in postUp/postdown id NatEnabled is leftout from the POST request when creating an Egress Gateway.

**Review by mattkasun — APPROVED** [review]

**Comment by theguy951357** [comment]
@Exchizz when NAT is off, I cannot ping the VPC on the egress.

**Comment by Exchizz** [comment]
> @Exchizz when NAT is off, I cannot ping the VPC on the egress.

One thing that come to my mind is, if you try to ping a VM in the VPC from a node in wireguard, you need to add a route to the VM in your VPC, so that it can route back the reply to the wireguard network. This is not needed when NAT is enabled :)

If that doesn't solve the problem, I would need a description of your setup so I can see if I can reproduce the error you're experiencing.

**Pull request merged**